### PR TITLE
Set job ids on the server side

### PR DESCRIPTION
**Pull request by smondet**

### CODE REVIEW

File: src/ketrew_backend/plugin.ml [hunk 1]
```diff
@@ -8,7 +8,7 @@ module Run_parameters = struct
 
   type t = [
     | `Created of Client.t * Kube_job.Specification.t
-    | `Running of Client.t * Kube_job.Specification.t
+    | `Running of Client.t * Kube_job.Specification.t * string
```
Comment (ihodes) [inline src/ketrew_backend/plugin.ml]: What about something like `type id = string`, or add a comment?

File: src/lib/kube_job.ml [hunk 2]
```diff
@@ -90,6 +85,11 @@ let command_must_succeed_with_output ~log ?additional_json job cmd =
 let start ~log t =
   let spec = t.specification in
   let open Specification in
+  let secret_name f =
+    String.take
+      (t.id ^ (Digest.string (File_contents_mount.show f) |> Digest.to_hex))
+      55 (* The max is 63 + we want to add "-volume" *)
```
Comment (ihodes) [inline src/lib/kube_job.ml]: how long is t.id?

File: src/ketrew_backend/plugin.ml (hunk 1 above)
Comment (smondet) [inline src/ketrew_backend/plugin.ml]: 👍

File: src/lib/kube_job.ml (hunk 2 above)
Comment (smondet) [inline src/lib/kube_job.ml]: it's UUID "V4" https://tools.ietf.org/html/rfc4122
http://erratique.ch/software/uuidm/doc/Uuidm

File: src/ketrew_backend/plugin.ml (hunk 1 above)
Comment (smondet) [inline src/ketrew_backend/plugin.ml]: In a record now :)

File: src/lib/kube_job.ml (hunk 2 above)
Comment (ihodes) [inline src/lib/kube_job.ml]: Got is, so 36 characters, leaving 19 for the mount digest. That should be fine :)

**Comment by ihodes** [comment]
LGTM!

**Pull request merged**